network runner docs nits (#443)
ava-labs/avalanche-docs: docs/build/tools/netrunner.md

# Network Runner

The Avalanche Network Runner tool allows a user to define, create and interact with a network of Avalanche nodes. It can be used for development and testing.

**Note that this tool is not for running production nodes.**

## Installation

> The Network Runner repository is hosted at [https://github.com/ava-labs/avalanche-network-runner.](https://github.com/ava-labs/avalanche-network-runner)

Reviewer: ```suggestion
The Network Runner repository is hosted at [https://github.com/ava-labs/avalanche-network-runner](https://github.com/ava-labs/avalanche-network-runner).
```

That repository's README details the tool.

Clone the repository with:

```
git clone https://github.com/ava-labs/avalanche-network-runner.git
```

> Except when otherwise specified, file paths given below are relative to the root of this repository. 

Reviewer: ```suggestion
Unless otherwise specified, file paths given below are relative to the root of this repository. 
```

## Usage

The basic pattern for using the Avalanche Network Runner is:

* Define the network
* Start the network
* Interact with it
* Shutdown

The [api](https://github.com/ava-labs/avalanche-network-runner/tree/main/api) package contains client code that allows the user to make API calls to Avalanche nodes within the network.

Please find examples of usage in the [examples](https://github.com/ava-labs/avalanche-network-runner/tree/main/examples) subdirectory.

## Supported Backends

As you can see in the Network Runner's README, it exposes a `Network` interface to the user. There are two implementations of this interface that you may use. One is the [local](https://github.com/ava-labs/avalanche-network-runner/tree/main/local) implementation, where each node runs in a process on your machine. The other is the [Kubernetes](https://github.com/ava-labs/avalanche-network-runner/tree/main/k8s) implementation, where each node runs in its own Kubernetes pod.

## Local Backend

> The simplest and most straightforward way to use the Network Runner is to use the local implementation. When you create a network with this tool, each node runs in a process on your machine. **Please note that the maximum size and performance of this network depend on your machine's specificaitons.**

Reviewer: ```suggestion
The simplest and most straightforward way to use the Network Runner is to use the local implementation. When you create a network with this tool, each node runs in a process on your machine. **Please note that the maximum size and performance of this network depend on your machine's specifications.**
```

> An example can be found at [examples/local/main.go](https://github.com/ava-labs/avalanche-network-runner/blob/main/examples/local/main.go). It is a simple setup which just creates a network of 5 nodes, and performs some simple API calls.

Reviewer: ```suggestion
An example can be found at [examples/local/main.go](https://github.com/ava-labs/avalanche-network-runner/blob/main/examples/local/main.go). It is a simple setup which creates a network of 5 nodes, and performs some simple API calls.
```

Please refer to [NetworkConfig](https://github.com/ava-labs/avalanche-network-runner#network-creation) for more details.

## Kubernetes Backend

It's possible to create a network with a Kubernetes backend, offering higher flexibility and scalability in defining and running a development and test network.

An example can be found at `examples/k8s/main.go`. This represents an example default network of 5 nodes which runs in a Kubernetes cluster.

Note that the Kubernetes backend should only be used by advanced users, and requires more setup to use. 

### AvalancheGo Operator

**IMPORTANT**:
In order for AvalancheGo nodes to run in a Kubernetes cluster, we use the [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/). Therefore, for an AvalancheGo network to be runnable on Kubernetes, the `avalanchego-operator` dependency must be fulfilled. Essentially, the operator allows AvalancheGo nodes to run inside a Kubernetes cluster in a stateful manner.

Please find the AvalancheGo Operator code at [https://github.com/ava-labs/avalanchego-operator.](https://github.com/ava-labs/avalanchego-operator)

> There are a multitude of possible configurations and setups when running Kubernetes, depending on your organization, preferences and environment. Generally, the Kubernetes admin should be responsible for installing and running the operator, as it involves setting up roles, services and permissions. Please refer to the AvalancheGo Operator documentation for instructions on installation, configuration, deployment, etc..

Reviewer: ```suggestion
There are a multitude of possible configurations and setups when running Kubernetes, depending on your organization, preferences and environment. Generally, the Kubernetes admin should be responsible for installing and running the operator, as it involves setting up roles, services and permissions. Please refer to the AvalancheGo Operator documentation for instructions on installation, configuration, deployment, etc.
```
